### PR TITLE
[wip] feat: Enable HTTP API Server

### PR DESCRIPTION
**Pull request by adityapk00**
## Motivation

Add documentation for the HTTP API server and enable it

## Change Summary

- Add docs

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [X] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [X] PR has a [changeset](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#35-adding-changesets)
- [ ] PR has been tagged with a change label(s) (i.e. documentation, feature, bugfix, or chore)
- [X] PR includes [documentation](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#32-writing-docs) if necessary.
- [X] All [commits have been signed](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#22-signing-commits)

---

### Focus of this PR:
Enable the HTTP API server in the Hubble application.

### Detailed summary:
- Added the feature to enable the HTTP API server.
- Updated the README.md file with information about using the HTTP API.
- Added a new link for HTTP APIs in the configuration file.
- Modified the linter script to include HTTP API documentation.
- Updated package.json to include new dependencies.
- Modified test files to include the new SyncEngine.
- Updated the documentation in hub-nodejs to mention the HTTP API.
- Modified the hubble.ts file to start and stop the HTTP API server.
- Updated the CLI documentation to include the option to enable/disable the admin server and HTTP server.

> The following files were skipped due to too many changes: `apps/hubble/www/docs/docs/cli.md`, `apps/hubble/src/cli.ts`, `apps/hubble/src/rpc/server.ts`, `apps/hubble/src/network/sync/merkleTrie.ts`, `apps/hubble/scripts/httpapidocs.js`, `apps/hubble/src/rpc/test/httpServer.test.ts`, `apps/hubble/src/rpc/httpServer.ts`, `apps/hubble/www/docs/docs/httpapi.md`, `yarn.lock`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/hubble/src/network/sync/merkleTrie.ts [hunk 1]
```diff
@@ -112,9 +112,16 @@ class MerkleTrie {
     this._worker.addListener("message", async (event) => {
       // console.log("Received message from worker thread", event);
       if (event.dbGetCallId) {
-        const value = await ResultAsync.fromPromise(this._db.get(Buffer.from(event.key)), (e) => e as Error);
-        if (value.isErr()) {
-          log.warn({ key: event.key, error: value.error }, "Error getting value from DB");
+        // This can happen sometimes in tests when the DB is closed before the worker thread
+        let value = undefined;
+        if (this._db.status === "closed") {
```
Comment (sanjayprabhu) [inline apps/hubble/src/network/sync/merkleTrie.ts:117]: These changes seem unrealted to the http server?

File: apps/hubble/src/hubble.ts [hunk 2]
```diff
@@ -575,6 +578,11 @@ export class Hub implements HubInterface {
 
     // Start the RPC server
     await this.rpcServer.start(this.options.rpcServerHost, this.options.rpcPort ?? 0);
+    if (!this.options.httpServerDisabled) {
+      await this.httpApiServer.start(this.options.rpcServerHost, this.options.httpApiPort ?? 0);
```
Comment (sanjayprabhu) [inline apps/hubble/src/hubble.ts:582]: What about auth? I think if someone enables RPC auth, they would not intend for the http server to remain accessible.

File: apps/hubble/scripts/httpapidocs.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+ * This linter checks that each of the HTTP API server endpoints is documented properly.
+ * This will check:
+ *   1. That all endpoints have a "@doc-tag:" comment in the httpServer.ts file
+ *   2. Make sure that all endpoints have a corresponding section in the HTTP API docs
+ *   3. Make sure that all parameters for every endpoint are documented in the HTTP API docs
+ *      under the corresponding section
+ *   4. Make sure that all parameters that are documented are specified in the @docs-tag comment
+ *      for that endpoint in the httpServer.ts file
+ */
+export function httpapidocs() {
```
Comment (sanjayprabhu) [inline apps/hubble/scripts/httpapidocs.js:21]: Nice, could we also add a check that all protobuf rpc methods have a corresponding http api call (unless explicitly marked unsupported)?

File: apps/hubble/src/network/sync/merkleTrie.ts (hunk 1 above)
Comment (adityapk00) [reply apps/hubble/src/network/sync/merkleTrie.ts:117]: The bug was exposed in the httpserver.test.ts, which is in this PR, so I decided to include it in here.

File: apps/hubble/src/hubble.ts (hunk 2 above)
Comment (adityapk00) [reply apps/hubble/src/hubble.ts:582]: The Auth will work with the HTTP API as well (although I've not implemented yet). Basically, you can have the same auth apply to both HTTP and gRPC API. Hence the separate flag.

File: apps/hubble/scripts/httpapidocs.js (hunk 3 above)
Comment (adityapk00) [reply apps/hubble/scripts/httpapidocs.js:21]: Hmm, this is a good idea. Let me see how to implement it!

**Pull request merged**